test/odbc/psqlodbc/xml.cpp

```diff
@@ -0,0 +1,230 @@
+
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Xml, Data_Length) {
+  vector<string> inserted_values = {
+    "NULL",
+    SIMPLE_XML,
+    NORMAL_XML,
+    LARGE_XML
+  };
+  const int number_of_entries = inserted_values.size();
+
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  testInsertionSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+  insertValuesInTable(ServerType::PSQL, PG_TABLE_NAME, inserted_values, false, number_of_entries);
+
+  inserted_values = duplicateElements(inserted_values);
+
+  verifyValuesInObject(ServerType::PSQL, PG_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+  verifyValuesInObject(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Xml, Update_Success) {
+  vector<string> inserted_values = {
+    NORMAL_XML
+  };
+
+  vector <string> data_update_values = {
+    SIMPLE_XML
+  };
+
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  testInsertionSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+
+  testUpdateSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, COL2_NAME, data_update_values, data_update_values);
+  verifyValuesInObject(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, data_update_values, data_update_values);
+
+  data_update_values = {
+    SIMPLE_XML
+  };
+
+  testUpdateSuccess(ServerType::PSQL, PG_TABLE_NAME, COL1_NAME, COL2_NAME, data_update_values, data_update_values);
+  verifyValuesInObject(ServerType::PSQL, PG_TABLE_NAME, COL1_NAME, data_update_values, data_update_values);
+
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Xml, View_Creation) {
+  const vector<string> INSERTED_VALUES = {
+    SIMPLE_XML,
+    NORMAL_XML
+  };
+  const int NUM_OF_INSERTS = INSERTED_VALUES.size();
+
+  const string BBF_VIEW_QUERY = "SELECT * FROM " + BBF_TABLE_NAME;
+  const string PG_VIEW_QUERY = "SELECT * FROM " + PG_TABLE_NAME;
+
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  testInsertionSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
+
+  createView(ServerType::MSSQL, BBF_VIEW_NAME, BBF_VIEW_QUERY);
+
+  verifyValuesInObject(ServerType::MSSQL, BBF_VIEW_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
+  verifyValuesInObject(ServerType::PSQL, PG_VIEW_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
+
+  dropObject(ServerType::MSSQL, "VIEW", BBF_VIEW_NAME);
+  dropObject(ServerType::PSQL, "VIEW", PG_VIEW_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+}
```

Comment:
Nit: Please add new line at end of file.

Reply:
Done.